framework/Sources/VGSFramework/UIElements/Scan/VGSScanController.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  VGSScanController.swift
+//  VGSFramework
+//
+//  Created by Dima on 06.01.2020.
+//  Copyright © 2020 VGS. All rights reserved.
+//
+
+import Foundation
+#if canImport(UIKit)
+import UIKit
+#endif
+
+@objc public protocol VGSScanControllerDelegate {
+    @objc optional func userDidFinishScan()
+    @objc optional func userDidCancelScan()
+    @objc optional func userDidSkipScan()
+    @objc optional func getFormForScanedField(name: String) -> VGSTextField?
```

Comment:
How about using an enum for textFiled and use in this case FiealType `.cardNumber`?

Reply:
Also think about it. So if we will hv 10+ scan providers, each of them scan different fields: passport number, age, etc. Then we should make large Enum to handle all cases. And if user use 1 scan provider he will use only 2-3 cases from our large Enum.